core/src/main/resources/org/apache/spark/ui/static/executorspage.js

```diff
@@ -172,6 +172,15 @@ function totalDurationColor(totalGCTime, totalDuration) {
     return (totalGCTime > GCTimePercent * totalDuration) ? "white" : "black";
 }
 
+function memoryUsageTooltip(onHeap, maxOnHeap, offHeap, maxOffHeap, onHeapSum, offHeapSum, type) {
+    return ("<span data-toggle='tooltip' title='" +
+        "On Heap Memory (" + formatBytes(onHeap, type) +
+        " / " + formatBytes(maxOnHeap, type) + ") " +
+        "Off Heap Memory (" + formatBytes(offHeap, type) +
+        " / " + formatBytes(maxOffHeap, type) + ")'>" +
+        formatBytes(onHeapSum, type) + " / " + formatBytes(offHeapSum, type) + "</span>");
```

Comment:
this is pretty hard to read (at least for me, though I don't really know js ...).  Maybe refactor so you prepare two tmp variables, one with the tooltip text, and another one w/ the normal text, and then return the fully assembled string?